### ISSUE TITLE
Possible idea to get rid of keys

### ISSUE THREAD

**Issue by tvld** [gadicc/meteor-messageformat#142]
I am working a few days with mf, I like it. But the keys make it a little less elegant in the code. However, I do understand the reasoning why we need it.

As meteor tends to work nicely with quite compact templates, I thought about this idea:

What if mf would keep the scope of keys within the context of a template, so we just can use a small integer as key? This int is unique for the template only so text can be rearranged without loosing translations. 

**Example** 

`<template name="home">`
`  {{mf 1 'Welcome'}}`
`  <p>{{mf 2 'This is nice to see you'}}</p>`
` </template>`

` <template name="apage">`
`  {{mf 16 'A page of us'}}`
`  <p>{{mf 2 'The quick brown fox....'}}</p>`
` </template>`

This would give parameters:
 *home_1*
 *home_2*
 *apage_2*
 *apage_16*`

**Parsing Logic**
- if *mf_extract* finds first (key) parameter of int, mf knows to treat keys as local, using the template name as prefix. 
- In cases where for some reason (dont know if they exist), there is no template name, mf_extract could use the file name as the context
- multiple use of a certain int in one context, will throw an error
- if the mf_extract finds a string, all logic is as is now

Thoughts?
Tom

**Comment by DSpeichert** [comment]
Thanks for your comments, Tom, it does sound like a good idea. There was an idea before to make the keys local to the template but as far as I remember, it was not possible a while ago for the mf_extract to discover what template the key was in (in a feasible way).

Things have probably changed so far so it's worth looking into. I'm sure @gadicc can say more

**Comment by gadicc** [comment]
Yeah, this is a very nice compromise.  As @DSpeichert mentioned, we had wanted template localized before, and enumerating them is a great idea.  The problem wasn't actually with mf_extract, but with figuring out what template we were being called from in Meteor.  At the time it wasn't possible but indeed now it is.  So let's leave this open and once we get passed a few more milestones on the v2 branch, it's something we can try out.  We also discussed some ideas for renaming keys in https://github.com/gadicc/meteor-messageformat/issues/105.

**Comment by tvld** [comment]
Thanks:)  As per your comments I've updated initial post to allow easier discussion & development later.

**Comment by gadicc** [comment]
Awesome, shot :)
